UI/win-update/updater/updater.cpp

```diff
@@ -594,6 +596,78 @@ static inline bool WideToUTF8(char *utf8, int utf8Size, const wchar_t *wide)
 				     nullptr, nullptr);
 }
 
+#define UTF8ToWideBuf(wide, utf8) UTF8ToWide(wide, _countof(wide), utf8)
+#define WideToUTF8Buf(utf8, wide) WideToUTF8(utf8, _countof(utf8), wide)
+
+/* ----------------------------------------------------------------------- */
+
+queue<string> hashQueue;
+
+void HasherThread()
+{
+	bool hasherThreadFailure = false;
+	unique_lock<mutex> ulock(updateMutex, defer_lock);
+
+	while (true) {
+		ulock.lock();
+		if (hashQueue.empty())
+			return;
+
+		auto fileName = hashQueue.front();
+		hashQueue.pop();
+
+		ulock.unlock();
+
+		wchar_t updateFileName[MAX_PATH];
+
+		if (!UTF8ToWideBuf(updateFileName, fileName.c_str()))
+			continue;
+		if (!IsSafeFilename(updateFileName))
+			continue;
+
+		BYTE existingHash[BLAKE2_HASH_LENGTH];
+		wchar_t fileHashStr[BLAKE2_HASH_STR_LENGTH];
+
+		if (CalculateFileHash(updateFileName, existingHash)) {
+			HashToString(existingHash, fileHashStr);
+			ulock.lock();
+			hashes.emplace(fileName, fileHashStr);
+			ulock.unlock();
+		}
+	}
+
+	return;
+}
+
+static void RunHasherWorkers(int num, const Json &packages)
+try {
+
+	for (const Json &package : packages.array_items()) {
+		for (const Json &file : package["files"].array_items()) {
+			if (!file["name"].is_string())
+				continue;
+			hashQueue.push(file["name"].string_value());
+		}
+	}
```

Comment:
I haven't included this, but I found that by reversing the order of packages (thus adding `obs-browser` first) I can actually shave off another 100 ms or so since CEF will be processed first rather than at the end, meaning that we spend less time with only a single thread working on CEF while the others are already finished.

```c++
	/* Okay I know this looks mad. But there's actually a somewhat sane reason
	 * to do this:
	 *
	 * libcef.dll takes the longest of all files to hash, so by ensuring the processing
	 * of it gets started as soon as possible we don't spend as long waiting for the last
	 * thread to complete as if it was added in its normal position (at the end).
	 *
	 * (Technically this will add any file > 10 MiB to the start, which is only three )
	 * 
	 * This saves a whole 50 ms on my machine!
	 */
	Json::array pVec = packages.array_items();
	for_each(pVec.rbegin(), pVec.rend(), [](const Json &package) {
		for (const Json &file : package["files"].array_items()) {
			if (!file["name"].is_string())
				continue;
			hashQueue.push(file["name"].string_value());
		}
	});
```